baseapp/baseapp.go

```diff
@@ -609,6 +609,141 @@ func (app *BaseApp) cacheTxContext(ctx sdk.Context, txBytes []byte) (sdk.Context
 	return ctx.WithMultiStore(msCache), msCache
 }
 
+// runCheckTxConcurrently processes a transaction with either the checkTx or recheckTx modes, encoded transaction
+// bytes, and the decoded transaction itself. All state transitions occur through
+// a cached Context depending on the mode provided.
+//
+// Note, gas execution info is always returned. A reference to a Result is
```

Comment:
> gas execution info is always returned ... tx does not run out of gas

Is this true? i think gas doesn't matter unless we run `DeliverTx` mode and I think we only run this w checkTx or recheckTx. This comment doesn't seem super relevant

Reply:
Yes since the [ABCI CheckTx response returns gas wanted/gas](https://github.com/dydxprotocol/cosmos-sdk/blob/6477dcf5693422fef693432e29bd979709aa995d/baseapp/abci.go#L391) used on it.